langchain-interpreter/src/langchain_interpreter/agent.py

```
@@ -0,0 +1,620 @@
+            # Update session history
+            session_history.append(HumanMessage(content=user_input))
+            session_history.append(AIMessage(content=response_content))
+
+            return result
+
+        except InputValidationError:
+            raise
+        except Exception as e:
+            if isinstance(e, AgentError):
+                raise
+            raise AgentError(f"Agent execution failed: {e}") from e
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)